service/resource/resourcegroup/resource.go

```diff
@@ -0,0 +1,289 @@
+package resourcegroup
```

Comment:
Again me being boring about packages flattening. `service/resource/resourcegroup/resource.go` it doesn't look good when using the code. Should be `service/resource/resource_group.go`. Then `resource.NewResourceGroup(config)`. We tend to make a lot of small packages and copy/paste code.

Reply:
I don't think flattening works in this case because each resource needs to implement the Resource interface. Also I think there will only be a small number of resources (Resource Group, Deployment and CloudConfig). For me having them in separate packages makes the code easier to understand.